contrib/offline/manage-offline-container-images.sh

```diff
@@ -1,14 +1,93 @@
 #!/bin/bash
 
 OPTION=$1
-CURRENT_DIR=$(cd $(dirname $0); pwd)
+CURRENT_DIR=$( dirname "$(readlink -f "$0")" )
 TEMP_DIR="${CURRENT_DIR}/temp"
-
 IMAGE_TAR_FILE="${CURRENT_DIR}/container-images.tar.gz"
 IMAGE_DIR="${CURRENT_DIR}/container-images"
 IMAGE_LIST="${IMAGE_DIR}/container-images.txt"
 RETRY_COUNT=5
 
+K8S_IMAGES_LIST_FILE=${K8S_IMAGES_LIST_FILE:-"${TEMP_DIR}/images.list"}
+K8S_IMAGE_DIR="${CURRENT_DIR}/k8s-images"
+K8S_IMAGE_INFO_FILE="${K8S_IMAGE_DIR}/images.info"
+K8S_IMAGE_ARCHIVE="${CURRENT_DIR}/k8s-images.tar.gz"
+
+function save_image_to_file() {
+  image=$1
+  info_file=$2
+  if [ -z "${image}" ]; then
+    echo "image should be provided."
+    exit 1
+  fi
+  if [ -z "${info_file}" ]; then
+    echo "info file should be provided."
+    exit 1
+  fi
+
+  FILE_NAME="$(echo ${image} | sed s@"/"@"-"@g | sed s/":"/"-"/g)".tar
+  [ -f "${FILE_NAME}" ] && return 0
+  set +e
+  for step in $(seq 1 ${RETRY_COUNT})
+  do
+    sudo docker pull ${image}
+    if [ $? -eq 0 ]; then
+      break
+    fi
+    echo "Failed to pull ${image} at step ${step}"
+    if [ ${step} -eq ${RETRY_COUNT} ]; then
+      exit 1
+    fi
+  done
+  set -e
+  sudo docker save -o "${FILE_NAME}"  "${image}"
+
+  # NOTE: Here removes the following repo parts from each image
+  # so that these parts will be replaced with Kubespray.
+  # - kube_image_repo: "registry.k8s.io"
+  # - gcr_image_repo: "gcr.io"
+  # - docker_image_repo: "docker.io"
+  # - quay_image_repo: "quay.io"
+  FIRST_PART=$(echo "${image}" | awk -F"/" '{print $1}')
+  if [ "${FIRST_PART}" = "registry.k8s.io" ] ||
+     [ "${FIRST_PART}" = "gcr.io" ] ||
+     [ "${FIRST_PART}" = "docker.io" ] ||
+     [ "${FIRST_PART}" = "quay.io" ] ||
+     [ "${FIRST_PART}" = "${PRIVATE_REGISTRY}" ]; then
+    image=$(echo "${image}" | sed s@"${FIRST_PART}/"@@)
+  fi
+  echo "${FILE_NAME}  ${image}" >> "${info_file}"
+}
+
+function archive_k8s_images() {
+
+  if [ ! -f "${K8S_IMAGES_LIST_FILE}" ]; then
+    echo "K8s image list file do NOT exist, run generate_list.sh first."
```

Comment:
To understand the issue easily
```suggestion
    echo "K8s image list file(${K8S_IMAGES_LIST_FILE}) do NOT exist, run generate_list.sh first."
```